### ISSUE TITLE
pygame.mixer.music --> rewind() during pause() causes no clean start from beginning (2501)

### ISSUE THREAD

**Issue by GalacticEmperor1** [pygame-community/pygame-ce#1288]
#  **Issue &numero;2501 opened by *[himijendrix24](https://github.com/himijendrix24)* at 2021-02-26 07:47:59**

**Environment:**

- Linux, raspbian
- python 3.7.3
- SDL 2.0.9
- pygame 2.0.1
- raspberry-pi zero + hifiberry dac+ zero

**Current behavior:**

In the pygame.mixer.music class, when calling rewind() after pause() and than play() again,
some samples after the last pause() position are played before starting from the beginning. 
It's really short but depending on the application annoying.

**Expected behavior:**

a clean playback from the beginning when calling rewind() during pause() followed by a play().
Instead you have to call rewind() during playing the file and right after that pause() which works but is not intuitive.

**Test code**

```
import pygame
pygame.mixer.init(frequency=48000, size=-16, channels=2, buffer=2048)
pygame.mixer.init()
pygame.mixer.music.load('/home/pi/test.mp3')

pygame.mixer.music.play()
pygame.mixer.music.pause()
pygame.mixer.music.rewind()
pygame.mixer.music.play() #  --> plays some left over samples first
```<hr>

#  Comments

# # #  **[Starbuck5](https://github.com/Starbuck5)* commented at 2021-02-26 08:41:35*

Thank you for the report.
I can reproduce this on my windows system.
It's a lot easier to hear if you put time.sleeps between the play and the pause, and the pause and the rewind.

> Instead you have to call rewind() during playing the file and right after that pause() which works but is not intuitive.

Does that really work? I didn't have luck with that myself.

Also this looks like an SDL_Mixer problem, so we might have to report it to them.

**Comment by zoldalma999** [comment]
Does not reproduce for me (windows), and I asked Ankith to test it for me, he says he can't hear it either anymore (ubuntu).

Seems like this was fixed between SDL_Mixer 2.6.2 (which is what current main uses) and SDL_Mixer 2.0.4 (which pygame 2.0.1 ships with). The issue does come back if you use mixer 2.0.4 again on current main. I could not find any explicit issue, pr, or release note section about this though.

Closing as fixed, but if someone does reproduce the original issue with a modern SDL_Mixer version, do comment or open another issue
